llama_cpp_guidance/llm.py

```diff
@@ -82,42 +86,46 @@ def session(self, asynchronous=False):
             raise NotImplementedError
 
     def __call__(self, *args, **kwargs):
-        logger.debug("Invoking LlamaCpp ({args}) ({kwargs})", args=args, kwargs=kwargs)
-        output = self.llm(*args, **kwargs)
-        logger.debug(
-            "LlamaCpp generated text: {text} ({choices})",
-            text=output["choices"][0]["text"],
-            choices=[
-                (choice["text"], choice["logprobs"]) for choice in output["choices"]
-            ],
-        )
+        print("Invoking LlamaCpp ({args}) ({kwargs})", args, kwargs)
 
-        for choice in output.get("choices", []):
-            logprobs = choice.get("logprobs")
+        llm_out = self.llm(*args, **kwargs)
+        if isinstance(llm_out, dict):
+            llm_out = [llm_out]
```

Comment:
This is a _way_ better way to handle Generator output that the previous solution. Fantastic update.